python-package/lightgbm/sklearn.py

```diff
@@ -635,6 +656,25 @@ def _process_params(self, stage: str) -> Dict[str, Any]:
         # overwrite default metric by explicitly set metric
         params = _choose_param_value("metric", params, original_metric)
 
+        # use joblib conventions for negative n_jobs, just like scikit-learn
+        n_jobs = self.n_jobs
+        if stage == "fit":
```

Comment:
Could you please clarify, why this `if` guard is needed here?
`n_jobs` can be passed for the prediction methods as well.

Reply:
`n_jobs` is always passed at `fit`, but not always passed at `predict`. Ommitting it at prediction time means using the default.

Reply:
Right now at prediction time two aliases (default `n_jobs` set in init of fit stage and manually calculated `num_threads`) of the same parameter are passed to cpp code which is incorrect.

Reply:
It's passing only one actually. But the issue is, `None` at fit-time means "use the number of cores", while at predict-time it means "use the default number of omp threads".